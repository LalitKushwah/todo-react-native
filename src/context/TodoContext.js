import React, {createContext, useState} from 'react';

const TodoContext = createContext();

const TodoContextProvider = props => {
  const [todos, setTodos] = useState([]);
  const addTodo = title => setTodos(todos => {
    const newTodo =  { title: title, id: new Date() }
    const updated = [...todos, newTodo];
    return updated;
  });


  const removeTodo = id =>
    setTodos(() => todos.filter(item => item.id !== id));

  return (
    <TodoContext.Provider value={{addTodo, removeTodo, todos}}>
      {props.children}
    </TodoContext.Provider>
  );
};

export {TodoContext, TodoContextProvider};
