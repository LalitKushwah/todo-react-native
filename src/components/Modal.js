import * as React from 'react';
import {useContext, useState} from 'react';
import {Modal, Portal, Button, Provider, TextInput} from 'react-native-paper';
import {TodoContext} from '../context/TodoContext';

const TodoModal = props => {
  const {addTodo} = useContext(TodoContext);
  const [textValue, setTextValue] = useState('');
  const containerStyle = {
    backgroundColor: 'white',
    padding: 20,
    width: '100%',
  };
  return (
    <Provider>
      <Portal>
        <Modal
          {...props}
          style={{alignItems: 'center', justifyContent: 'center'}}
          contentContainerStyle={containerStyle}
          testID='todo-modal'>
          <TextInput
            label="Enter Todo here"
            value={textValue}
            onChangeText={setTextValue}
            testID='modal-text-input'
          />
          <Button
            mode="contained"
            style={{marginTop: 30}}
            onPress={() => {
              addTodo(textValue);
              props.onDismiss();
            }}
            disabled={!textValue}
            testID='action-btn'>
            Add Todo
          </Button>
        </Modal>
      </Portal>
    </Provider>
  );
};

export default TodoModal;
