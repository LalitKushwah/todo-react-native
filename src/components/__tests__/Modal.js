import React from 'react';
import {render } from '@testing-library/react-native';
import Modal from '~/components/Modal';
import {TodoContextProvider} from '../../context/TodoContext';

describe('<TodoList/>', () => {
  test('it should capture snapshot of component', () => {
    const tree = render(<TodoContextProvider><Modal /></TodoContextProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
