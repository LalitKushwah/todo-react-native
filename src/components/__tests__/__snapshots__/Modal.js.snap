// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList/> it should capture snapshot of component 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
